src/raftstore/store/peer.rs

```diff
@@ -1884,6 +1884,9 @@ impl Peer {
         if self.is_merging() {
             return Err(box_err!("can not read index due to merge"));
         }
+        if !self.is_leader() && self.leader_id() == INVALID_ID {
```

Comment:
How did you find this bug? Better to cover it in schrodinger tests.

Reply:
if the peer is not leader you can return error directly, why it must be equal to 0?

Reply:
> if the peer is not leader you can return error directly, why it must be equal to 0?

A replica will set `leader_id` to `INVALID_ID` when election timeout (which shows no leader).

Reply:
Could you add a reference of how raft behaves on readindex if there is no leader?